stage0/src/zero_page.rs

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::fw_cfg::FwCfg;
+use core::{
+    ffi::CStr,
+    mem::{size_of, MaybeUninit},
+};
+use oak_linux_boot_params::{BootE820Entry, BootParams};
+
+#[link_section = ".boot.zero_page"]
+static mut BOOT_ZERO_PAGE: MaybeUninit<BootParams> = MaybeUninit::uninit();
+
+pub fn init_zero_page(fw_cfg: &mut FwCfg) -> &'static mut BootParams {
+    let mut zero_page = unsafe { BOOT_ZERO_PAGE.write(core::mem::zeroed()) };
+
+    // Magic constants.
+    zero_page.hdr.type_of_loader = 0xFF; // loader type undefined
+    zero_page.hdr.boot_flag = 0xAA55; // magic number
+    zero_page.hdr.header = 0x53726448; // Magic "HdrS" string
+    zero_page.hdr.kernel_alignment = 0x1000000; // Magic number from crosvm source.
```

Comment:
If you have a permalink to somewhere that defines these values, I think it will be useful to add them here as comments.

Reply:
Done.